### DISCUSSION TITLE
Plans for a replacement of <FullscreenModal />

### DISCUSSION THREAD

**Discussion by vankhoawin** [twilio-labs/paste#701] [Q&A] [answered]
Hey team! As we start migrating more modals to Paste, one case that's not covered is the FullscreenModal. In some cases, the [wide modal](https://paste.twilio.design/components/modal/#wide-modal) works fine. But in others, there is a need to stay in a page's context yet give users the ability to input data in a different context.

This might end up being more of a design system/UX question -- are there any replacements for the FullscreenModal component, component-wise or UX-wise?

**Comment by kate-porter** [comment, marked as answer]
Hey @vnguyen94! We do plan to have a full-screen modal in Paste, and it is _very tentatively_ planned in our November release right now. You can always check out our roadmap [here](https://paste-git-feat-websiteicon-page.twilio-dsys.vercel.app/roadmap/), but with the caveat that the further out the release date, the more tentative the list of components in that release.

Is your team in need of a full-screen modal soon? If so, could you share some more context on the use case and timeline? That will help us prioritize it.

Thank you!

**Reply by vankhoawin** [reply]
~Maybe I'm not reading it right, I don't see any mentions of modal for the November release?~ Aha I wasn't, I didn't expand the rest 😓 

And no, it's not a blocking issue (as much as say Tables), this is just to get a gauge of the work remaining to fully migrate out to Paste. It would be great to get a JIRA ticket so that we can link it as a blocker for PNREG-4837

**Reply by kate-porter** [reply]
Ha, all good!

And that makes sense, thanks for the context. We don't have the tickets written yet, but I just created an [empty epic](https://issues.corp.twilio.com/browse/DSYS-2240) for it so you can link to that in the meantime. Let me know if that works okay for ya.

**Reply by vankhoawin** [reply]
that's perfect, thanks Katie :) marking this as resolved.

**Reply by kate-porter** [reply]
Thanks @vnguyen94!

I saw on PNREG-4837 that you have 6 FullscreenModal implementations currently. If it's not too much to ask, would you be able to provide URLs and/or screenshots of those instances? That would be really helpful for us when we start to define requirements for the new full-screen modal component for Paste.

If you don't have time to do that right now, that's also totally fine. Regardless, I'll include a link to this discussion on our Full-screen Modal epic with a note to circle back with you to get a better understanding of your current FullscreenModal implementations if that sounds good!

**Reply by vankhoawin** [reply]
2 of the modals will be deprecated, so there are actually 4 Fullscreen modals (will update our tickets accordingly). Especially note the "Regulatory Compliance Supporting Documents Modal" screenshot as there is lots of data to view/edit that would be too much for just the wide Modal.

note: none of the screenshots should include PII data :) 

**1. Emergency Calling (E911) Modal, accessed via the IncomingPhoneNumberInstance page**
**https://www.twilio.com/console/phone-numbers/:phoneNumberSid**

![image](https://user-images.githubusercontent.com/6936736/91749968-27d30d80-eb77-11ea-9a11-c90a024a9556.png)

![image](https://user-images.githubusercontent.com/6936736/91749946-20136900-eb77-11ea-916a-a4bc9721e0b1.png)

**2. Buy a Number Modal**
https://www.twilio.com/console/phone-numbers/search

![image](https://user-images.githubusercontent.com/6936736/91750187-86988700-eb77-11ea-9c01-0468223d462c.png)

**3. Regulatory Compliance Supporting Documents Modal**
https://www.twilio.com/console/phone-numbers/regulatory-compliance/bundles/:bundleSid

![image](https://user-images.githubusercontent.com/6936736/91750347-d0816d00-eb77-11ea-8428-1b5ba8480291.png)

**4. Regulatory Compliance End-Users Modal**
https://www.twilio.com/console/phone-numbers/regulatory-compliance/bundles/:bundleSid

![image](https://user-images.githubusercontent.com/6936736/91750424-eee76880-eb77-11ea-9b33-af16f9ea0cb4.png)

**Reply by kate-porter** [reply]
This is _so_ helpful. Thank you!
